src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1508,28 +1557,199 @@ abstract void onApplication(
+                long affinity)
+            {
+                assert state == 0;
+                state = MqttState.openingInitial(state);
+
+                this.signalFuture = executor.schedule(publishTimeout, SECONDS, routeId, initialId,
+                        PUBLISH_TIMEOUT_SIGNAL);
+
+                router.setThrottle(initialId, this::onApplicationReply);
+                activePublishers.compute(topicFilter, (key, value) -> value != null ? value++ : 1);
+
+                doBegin(application, routeId, initialId, traceId, authorization, affinity, EMPTY_OCTETS);
+            }
+
+            private void doMqttDataEx(
+                long traceId,
+                long authorization,
+                OctetsFW payload,
+                Flyweight extension)
+            {
+                assert MqttState.initialOpening(state);
+
+                refreshPublishTimeout();
+
+                doData(application, routeId, initialId, traceId, authorization, 0L, payload.sizeof(),
+                        payload.buffer(), payload.offset(), payload.sizeof(), extension);
+            }
+
+            private void doInitialEnd(
+                long traceId,
+                long authorization,
+                Flyweight extension)
+            {
+                if (!MqttState.initialOpened(state) || initialSlot != NO_SLOT)
+                {
+                    state = MqttState.closingInitial(state);
+                }
+                else
+                {
+                    flushInitialEnd(traceId, authorization, extension);
+                }
+            }
+
+            private void doInitialAbort(
+                long traceId,
+                long authorization,
+                Flyweight extension)
+            {
+                setInitialClosed();
+
+                doAbort(application, routeId, initialId, traceId, authorization, extension);
+            }
+
+            private void doInitialAbortIfNecessary(
+                long traceId,
+                long authorization)
+            {
+                if (!MqttState.initialClosed(state))
+                {
+                    doInitialAbort(traceId, authorization, EMPTY_OCTETS);
+                }
+            }
+
+            private void flushInitialEnd(
+                long traceId,
+                long authorization,
+                Flyweight extension)
+            {
+                setInitialClosed();
+                doEnd(application, routeId, initialId, traceId, authorization, extension);
+            }
+
+            private void setInitialClosed()
+            {
+                assert !MqttState.initialClosed(state);
+
+                state = MqttState.closeInitial(state);
+                cleanupInitialSlotIfNecessary();
+
+                if (MqttState.closed(state))
+                {
+                    publishers.remove(topicKey(topicFilter));
+                    activePublishers.computeIfPresent(topicFilter, (key, value) -> value--);
```

Comment:
If value is `MutableInteger` this can change to
```
final MutableInteger count = activePublishers.get(topicFilter);
assert count != null;
count.value--;
assert value >= 0;
if (value == 0)
{
    activePublishers.remove(topicFilter);
}
```